packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -367,29 +362,17 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 		);
 	}
 
-	/**
-	 * Transition online blobs waiting for BlobAttach op round-trip since we will not see the op until we are connected
-	 * again
-	 */
-	private onDisconnected() {
-		for (const [localId, entry] of this.pendingBlobs) {
-			if (entry.status === PendingBlobStatus.OnlinePendingOp) {
-				// This will submit another BlobAttach op for this blob. This is necessary because the one we sent
-				// already didn't have the local ID.
-				this.transitionToOffline(localId);
-			}
-		}
-	}
-
 	private async shutdownPendingBlobs(): Promise<void> {
 		for (const [localId, entry] of this.pendingBlobs) {
-			if (entry.status === PendingBlobStatus.OnlinePendingUpload) {
-				this.sendBlobAttachOp(localId, entry.storageId);
-				entry.status = PendingBlobStatus.OfflinePendingUpload;
-				entry.handleP.resolve(this.getBlobHandle(localId));
-			} else if (entry.status === PendingBlobStatus.OnlinePendingOp) {
-				entry.status = PendingBlobStatus.OfflinePendingOp;
-				entry.handleP.resolve(this.getBlobHandle(localId));
+			switch (entry.status) {
+				case PendingBlobStatus.Uploading:
+					entry.status = PendingBlobStatus.Stashed;
```

Comment:
i feel like there are still to many concepts which makes this code hard to understand, stashed, pending, and shutdown? should this maybe become stash pending blobs?